authz/sdk_server_interceptors.go

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package authz
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/internal/xds/rbac"
+	"google.golang.org/grpc/status"
+)
+
+type StaticInterceptor struct {
+	// Either contains two engines deny engine followed by allow engine or only one allow engine.
+	engines rbac.ChainEngine
+}
+
+// NewStatic returns a new StaticInterceptor from a static authorization policy JSON string.
+func NewStatic(authzPolicy string) (*StaticInterceptor, error) {
+	RBACPolicies, err := translatePolicy(authzPolicy)
+	if err != nil {
+		return nil, err
+	}
+	chainEngine, err := rbac.NewChainEngine(RBACPolicies)
+	if err != nil {
+		return nil, err
+	}
+	if chainEngine.IsEmpty() {
+		return nil, fmt.Errorf("failed to initialize RBAC engines")
+	}
+	return &StaticInterceptor{*chainEngine}, nil
+}
+
+// UnaryInterceptor intercepts incoming Unary RPC request.
+// Only authorized requests are allowed to pass. Otherwise, unauthorized error is returned to client.
+func (i *StaticInterceptor) UnaryInterceptor(ctx context.Context, req interface{}, _ *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+	if err := i.engines.IsAuthorized(ctx); status.Code(err) != codes.OK {
+		return nil, err
```

Comment:
@dfawley I had a question regarding the errors that we return from rbac engine, which is then propagated to the client through here. We have errors like "incoming RPC matched a deny policy ..". I think we should have more general errors instead like "Unauthorized RPC request rejected" and maybe we can have logs for the details in rbac engine. Or even here in the interceptor I could return general error message. Wdyt?

Reply:
> Or even here in the interceptor I could return general error message. Wdyt?

I think it makes more sense to remove the details and log here, if that's appropriate, instead of doing those things in the engine itself.  Whether it's okay to return security policy deny reasons to the user, I'll leave that decision up to the security team -- we should do the same thing in all languages, too.

Reply:
Sounds good. In C++ we return "unauthorized RPC request rejected", so I will do the same for Go.